### PR TITLE
Add `misused-new` rule

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [X] Addresses an existing issue: #261
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [X] Documentation update
- [X] Enable CircleCI for your fork (https://circleci.com/add-projects)

#### What changes did you make?

Added the `misused-new` rule, which warns on attempts to define constructable interfaces and to use `new` in classes. This has come up a few times on [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

### CODE REVIEW

File: docs/rules/arrow-parens/index.html [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
+  if `ban-single-arg-parens` is specified, then arrow functions with one parameter
   must not have parentheses if removing them is allowed by TypeScript.
 options:
   type: string
   enum:
     - ban-single-arg-parens
 optionExamples:
   - 'true'
-  - '[true, "ban-single-arg-parens"]'
+  - '[true, "ban-single-arg-parens]"'
```
Comment (dphilipson) [inline docs/rules/arrow-parens/index.html]: This doesn't look intentional.

Comment (andy-hanson) [inline docs/rules/arrow-parens/index.html]: `docs` is generated, so this is due to an unrelated change (#1947). Made fix in #1967 and filed  #1968.

File: test/rules/misused-new/tslint.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "rules": {
+    "misused-new": true
```
Comment (adidahiya) [inline test/rules/misused-new/tslint.json]: let's call this `no-misused-new` to adhere better to existing naming conventions (we usually prefix "no-" to rules that lint _against_ the existence of some code)

**Comment by nchen63** [comment]
@andy-hanson thanks!
